proposals/array_modify.json

```
@@ -57,7 +57,7 @@
     },
     "exceptions": {
         "ArrayElementNotAvailable": {
-            "message": "The array has no element with the specified index."
+            "message": "An invalid insert index is specified."
```

I don't think this improves the message. Maybe:

```suggestion
            "message": "The array can't be modified as the given index doesn't exist."
```

---

First of all some context why I changed it:  if your given array has N elements, the range of valid indexes for `array_modify` is 0 to N (inclusive), so N+1 index values in total.  Index `N` is valid for `array_modify`, while there is no element there. The original message is bit confusing about that.

> I initially had "or negative" appended, but that should already be captured by the schema validation (minimum set to 0).

Yeah, but we want to throw that error `ArrayElementNotAvailable`  also on that schema violation. Or should we throw a generic  `ProcessParameterInvalid` ?

---

I'm not strictly against adding "or negative" again, just found it to be superfluous. So, yes, consistent would be if values that fails due to schema restrictions would return a ProcessParameterInvalid and runtime specific issues would return a specific error (here: ArrayElementNotAvailable).

---

I've changed to the longer error message without "or negative", but was not really sure whether to add it or not. Open for changes.